### PR TITLE
feat(treeview): tree view node actions allows dropdowns and buttons

### LINKED ISSUE #3518: Update tree view node action to allow dropdowns
re: https://github.com/patternfly/patternfly-react/issues/4799

Can we make the action element for the tree view be a generic container that allows buttons, dropdowns, etc?

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #3518

### CODE REVIEW

**Review by mattnolting — COMMENTED** [review]
It looks like adding a `dropdown` to `pf-c-tree-view__action` increases the height of the parent `node`.

Screen Shot 2020-09-29 at 11 27 50 AM
vs
Screen Shot 2020-09-29 at 11 28 38 AM

If we accomodate `pf-c-dropdown`, I think it's reasonable to assume that we should support other actionable components (like select). To do that, we either need to add individual components to this [css block](https://github.com/patternfly/patternfly/pull/3522/files#diff-b8ca8597c0e20fb6e7ef4aaad64f60b2R162) or apply the negative margining to `pf-c-tree-view__action` directly. 

@christiemolloy @mcoker wdyt?

**Comment by christiemolloy** [comment]
@mattnolting great catch! I think applying a negative margin to the actions element would be most flexible just in case more components try to get added

**Comment by christiemolloy** [comment]
Updated @mcoker and @mattnolting

File: src/patternfly/components/TreeView/tree-view.scss [hunk 1]
```diff
@@ -157,6 +160,10 @@ $pf-c-tree-view--MaxNesting: 10;
 
     background-color: var(--pf-c-tree-view__action--focus--BackgroundColor);
   }
+
+  > * {
+    margin-right: var(--pf-c-tree-view__action--child--MarginRight);
```
Comment (mcoker) [inline src/patternfly/components/TreeView/tree-view.scss:165]: Should this apply to `.pf-c-tree-view__action` instead?

File: src/patternfly/components/TreeView/examples/TreeView.md [hunk 2]
```diff
@@ -448,7 +455,6 @@ beta: true
 | `.pf-c-tree-view__node-text` | `<span>` | Initiates tree view text. |
 | `.pf-c-tree-view__node-icon` | `<span>` | Initiates a tree view icon. |
 | `.pf-c-tree-view__node-check` | `<span>` | Initiates a tree view check. |
-| `.pf-c-tree-view__action` | `<button>` | Initiates a tree view action. |
-| `.pf-c-tree-view__action-item` | `<span>` | Initiates a tree view action item. |
+| `.pf-c-tree-view__action` | `<div>` | Initiates a tree view action. |
```
Comment (mcoker) [inline src/patternfly/components/TreeView/examples/TreeView.md:458]: maybe this?

```suggestion
| `.pf-c-tree-view__action` | `<div>` | Initiates a tree view action wrapper. |
```

File: src/patternfly/components/TreeView/examples/TreeView.md [hunk 3]
```diff
@@ -402,15 +408,16 @@ beta: true
       {{/tree-view-list}}
     {{/tree-view-list-item}}
     {{#> tree-view-list-item tree-view-list-item--IsExpandable="true"}}
-      {{> tree-view-node tree-view-node--HasToggle="true" tree-view-node--text="Cost management"}}
+      {{> tree-view-node tree-view-node--type="div" tree-view-node--attribute='tabindex="0"' tree-view-node--HasToggle="true" tree-view-node--type="div" tree-view-node--text="Cost management"}}
     {{/tree-view-list-item}}
     {{#> tree-view-list-item tree-view-list-item--IsExpandable="true"}}
-      {{> tree-view-node tree-view-node--HasToggle="true" tree-view-node--text="Sources"}}
+      {{> tree-view-node tree-view-node--type="div" tree-view-node--attribute='tabindex="0"' tree-view-node--HasToggle="true" tree-view-node--type="div" tree-view-node--text="Sources"}}
```
Comment (mcoker) [inline src/patternfly/components/TreeView/examples/TreeView.md:414]: I wonder if we could build all of this into `tree-view-node--HasToggle` or any other attributes that make this need to be a div with a tabindex, etc.

Also you have `tree-view-node--type="div"` twice in this and some others.

**Comment by christiemolloy** [comment]
@mcoker updated

**Review by mattnolting — APPROVED** [review]
Perfecto!

**Review by mcoker — APPROVED** [review]
nice!

**Pull request merged**